Lib/test/test_concurrent_futures/test_process_pool.py

```diff
@@ -218,6 +228,81 @@ def mock_start_new_thread(func, *args, **kwargs):
                     list(executor.map(mul, [(2, 3)] * 10))
             executor.shutdown()
 
+    def test_process_pool_executor_terminate_kill_workers(self):
+        for function_name in ('terminate_workers', 'kill_workers'):
+            with self.subTest(function_name=function_name):
+                manager = multiprocessing.Manager()
+                q = manager.Queue()
+
+                with futures.ProcessPoolExecutor(max_workers=1) as executor:
+                    executor.submit(_put_sleep_put, q)
+
+                    # We should get started, but not finished since we'll terminate the workers just after
+                    self.assertEqual(q.get(timeout=5), 'started')
+
+                    worker_process = list(executor._processes.values())[0]
+                    getattr(executor, function_name)()
+                    worker_process.join()
+
+                    if function_name == 'terminate_workers' or sys.platform == 'win32':
```

Comment:
Instead of a loop you could refactor that test into a method which takes the function name and the expected exit code:

```py
def tesr_terminate(self):
    exit_code = ...
    self.do_test("terminate_workers", exit_code)
```

The other tests can also be exempt of a loop to reduce the indentation level but since they are already generic it's fine to keep them as is.

Reply:
I did something similar, yet also different since i found an existing parameterize fixture in the test suite already.